app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -118,32 +120,112 @@ class CustomChatInput extends ConsumerWidget {
             ),
           ),
         ),
-        Container(
-          width: double.infinity,
-          padding: const EdgeInsets.symmetric(vertical: 15),
-          color: Theme.of(context).colorScheme.onPrimary,
-          child: Center(
-            child: Padding(
-              padding: const EdgeInsets.symmetric(horizontal: 10),
-              child: Row(
-                mainAxisSize: MainAxisSize.min,
-                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  const _BuildAttachmentBtn(),
-                  const Expanded(
-                    child: Padding(
-                      padding: EdgeInsets.symmetric(horizontal: 10),
-                      child: _TextInputWidget(),
+        Visibility(
+          visible:
+              !ref.watch(chatInputProvider.select((ci) => ci.emojiRowVisible)),
+          replacement: Container(
```

Comment:
Can you explain to my your preference for the `Visibility`-Class rather than just toggling between the two potential states. For me, the visibility-class and the additional indentations make it less legible.

```suggestion
        !ref.watch(chatInputProvider.select((ci) => ci.emojiRowVisible)) ?  Container(
```


Also not sure if we have more "complex" sub widgets like this with their own state if they shouldn't be either an in-lined `Consumer` or better even be an externalized widget-class ...

This doesn't necessarily be addressed/changed in this PR but we might want to come to some agreement when to do what and thus I'd like to understand the reasoning behind using `Visibility`...

Reply:
I guess this is been reformed from legacy-code, where it was written to be ```Visibility```. There's no reasoning behind using ```Visibility``` other than toggling between states, which I would also argue that using ```Visibility``` seems to be more in-line with the UI widget pattern and understandable than using conditional.